Gems/AtomLyIntegration/CommonFeatures/Code/Source/ImageBasedLights/ImageBasedLightComponentController.cpp

```diff
@@ -131,22 +131,35 @@ namespace AZ
 
         void ImageBasedLightComponentController::UpdateWithAsset(Data::Asset<Data::AssetData> updatedAsset)
         {
-            if (m_configuration.m_specularImageAsset.GetId() == updatedAsset.GetId())
+            // REMARK: This function is typically invoked within the context of one of the AssetBus::OnAssetXXX functions,
+            // which may hold a mutex, in turn when instantiating StreamingImages, which is a multi-threaded operation,
+            // a deadlock may arise when the main thread is waiting on one of those threads to complete, but when
+            // one of those threads releases another StreamingImage object, that class executes
+            // Data::AssetBus::MultiHandler::BusDisconnect(GetAssetId()); which will acquire the same mutex of the original
+            // AssetBus::OnAssetXXX and triggers a deadlock.
+            // The solution is to enqueue texture update on the next tick.
```

Comment:
I have difficulty understanding this comment. Is this assessment correct?
1. AssetBus::OnAssetXXX is called (on the main thread?)
2. AssetBus::OnAssetXXX holds a mutex
3. OnAssetXXX instantiaties StreamingImages
4. Treads are spawned for instantiating StreamingImages
5. AssetBus::OnAssetXXX waits for these threads to complete
6. One of these threads releases another StreamingImage object
7. This calls Data::AssetBus::MultiHandler::BusDisconnect, which tries to acquire the mutex from 2.
8. Deadlock

If this is correct, could you write it out a bit more sequentially so it's easier to follow? Mainly thinking about someone else who might debug something in the future and will need to understand what problem we solved here with this design.
Thanks!

Reply:
You are correct. Will improve the comments.